code/game/gamemodes/malfunction/newmalf_ability_trees/HARDWARE.dm

```diff
@@ -87,19 +88,33 @@
 		return
 	if(!ability_prechecks(user, 0, 0))
 		return
+
+	//Lets find the first self destruct terminal
+	for(var/obj/machinery/nuclearbomb/station/N in SSmachinery.all_machines)
+		nuke = WEAKREF(N)
+		continue
+	
+	if(!nuke.resolve())
+		user << "Self-destruct could not be initiated - No Self-Destruct Terminal available."
+		return
+
 	user << "***** STATION SELF-DESTRUCT SEQUENCE INITIATED *****"
-	user << "Self-destructing in 2 minutes. Use this command again to abort."
+	user << "Self-destructing in 20 minutes. Use this command again to abort."
 	user.bombing_station = 1
 	set_security_level("delta")
-	radio.autosay("Self destruct sequence has been activated. Self-destructing in 120 seconds.", "Self-Destruct Control")
+	radio.autosay("Self destruct sequence has been activated. Self-destructing in 1200 seconds.", "Self-Destruct Control")
```

Comment:
Random thought. Make this a variable in the malf antag datum? Less shit to change if we ever want to change it.

Reply:
I had to change it back to be stored in the HARDWARE.dm again, because storing it in the datum would require the gamemode to be malf for the AI to use the nuke.
However that is not guaranteed since a malf AI can also be admin spawned in any gamemode.
Then the values for the time would be missing.